src/main/java/org/springframework/data/repository/config/DefaultImplementationLookupConfiguration.java

```diff
@@ -55,8 +56,7 @@
 
 		this.config = config;
 		this.interfaceName = interfaceName;
-		this.beanName = Introspector
-				.decapitalize(ClassUtils.getShortName(interfaceName).concat(config.getImplementationPostfix()));
+		this.beanName = Introspector.decapitalize(getLocalName(interfaceName).concat(config.getImplementationPostfix()));
```

Comment:
We should keep the implementation bean name as-is. Switching to the local name can clash with other nested beans and it introduces an inconsistency with fragment interfaces and nested repository bean names that are qualified with their enclosing class.

We can sort out that issue during the merge.